src/main/java/com/coveo/pushapiclient/DocumentUploadQueue.java

```diff
@@ -8,11 +8,11 @@
 /** Represents a queue for uploading documents using a specified upload strategy */
 class DocumentUploadQueue {
   private static final Logger logger = LogManager.getLogger(DocumentUploadQueue.class);
-  private final UploadStrategy uploader;
-  private final int maxQueueSize = 5 * 1024 * 1024;
-  private ArrayList<DocumentBuilder> documentToAddList;
-  private ArrayList<DeleteDocument> documentToDeleteList;
-  private int size;
+  protected final UploadStrategy uploader;
+  protected final int maxQueueSize = 5 * 1024 * 1024;
```

Comment:
With the batch size at [256MB](https://docs.coveo.com/en/lb4a0344/coveo-for-commerce/stream-your-catalog-data-to-your-source#limits), are we missing on some of the benefits of batching by setting it to only 5MB?

Reply:
Batch size is a weird topic to go into. Yes the batch size CAN be 256MB, but should it be is a bit harder to say.
From a purely performance standpoint, a batch size between 8 and 16MB is optimal, so aiming for that would be nice.

If the UPDATE purely consists of `AddOrUpdate` with no partials, then a batch size of 256MB is fine, although there might be an argument to do smaller batches so more workers can rip through the files.

The big issue with batch size comes with partials. We had a test source in the past where we had 10k documents, but each document is 1MB in size. This is a "small" amount of documents considering other catalog sources we have in prod. After that I sent an UPDATE with a partial update for each document. The file I uploaded to S3 was 950KB, well under our 256MB limit. This UPDATE took 40+ minutes to process since it needed to load/update/send every single document, meaning it caused 10+GB of IO traffic at minimum.

So what benefits are we missing on not setting it to 256MB? It depends on your workflow.

Reply:
Thinking about it more, this limit is probably more for the user. Im not sure this code would handle well if there was multiple thousands of documents of 1MB documents being passed though here with the 256MB limit. There's a lack of streaming and it's mainly just bulk serializing to json.

Reply:
I'm happy to look this at some point, but it goes beyond the scope of this change. If we do want to modify this then let's raise a ticket for the backlog